balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -575,6 +607,7 @@ func (b *pickfirstBalancer) updateSubConnState(sd *scData, newState balancer.Sub
 		// the first address when the picker is used.
 		b.shutdownRemainingLocked(sd)
 		b.state = connectivity.Idle
+		pickFirstDisconnectionsMetric.Record(b.metricsRecorder, 1, b.target)
```

Comment:
"The second condition above is considered a successful connection (again, as specified in the comment)." - I discussed this with Doug before I wrote this, and even though there's a interspliced successful connection, it can be logically be treated as simply a disconnection.

@dfawley

Reply:
For the second case, accounted for the interspliced READY signal by recording on connection attempts succeeded as well as disconnections.